### PR TITLE
moved data_dir to within `data/__init__.py`

### PR DESCRIPTION
**Pull request by hmaarrfk**
Added a small import for it back into `__init__.py`

## Checklist
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by jni** [comment]
I don't understand why but this appears to break our packaging quite severely!

**Comment by hmaarrfk** [comment]
Because it probably depends on #3158

**Comment by jni** [comment]
... Why? I don't see the relation at all? If master builds fine, why should these changes break it?

File: skimage/data/__init__.py [hunk 1]
```diff
@@ -12,13 +12,16 @@
 
 import numpy as _np
 
-from .. import data_dir
 from ..io import imread, use_plugin
 from .._shared._warnings import expected_warnings, warn
-from .. import img_as_bool
+from ..util.dtype import img_as_bool
```
Comment (soupault) [inline skimage/data/__init__.py]: Try reverting this line. It might cause circular import.

**Comment by hmaarrfk** [comment]
So as I was developing the original omnibus PR (which I admit was an overstep), I found that there was a circular import that caused things to fail. This seems to be the same thing:

```
C:\Users\mark\git\scikit-image>python setup.py develop
C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\setuptools\dist.py:388: UserWarning: Normalizing '0.15dev' to '0.15.dev0'
  normalized_version,
Partial import of skimage during the build process.
Traceback (most recent call last):
  File "setup.py", line 161, in <module>
    **extra
  File "C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\numpy\distutils\core.py", line 135, in setup
    config = configuration()
  File "setup.py", line 81, in configuration
    config.add_subpackage('skimage')
  File "C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\numpy\distutils\misc_util.py", line 1024, in add_subpackage
    caller_level = 2)
  File "C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\numpy\distutils\misc_util.py", line 993, in get_subpackage
    caller_level = caller_level + 1)
  File "C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\numpy\distutils\misc_util.py", line 930, in _get_configuration_from_setup_py
    config = setup_module.configuration(*args)
  File "skimage\setup.py", line 9, in configuration
    config.add_subpackage('_shared')
  File "C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\numpy\distutils\misc_util.py", line 1024, in add_subpackage
    caller_level = 2)
  File "C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\numpy\distutils\misc_util.py", line 993, in get_subpackage
    caller_level = caller_level + 1)
  File "C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\numpy\distutils\misc_util.py", line 906, in _get_configuration_from_setup_py
    ('.py', 'U', 1))
  File "C:\Users\mark\Miniconda3\envs\owl\lib\site-packages\numpy\compat\py3k.py", line 122, in npy_load_module
    return importlib.machinery.SourceFileLoader(name, fn).load_module()
  File "<frozen importlib._bootstrap_external>", line 399, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 823, in load_module
  File "<frozen importlib._bootstrap_external>", line 682, in load_module
  File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 684, in _load
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "skimage\_shared\setup.py", line 5, in <module>
    from skimage._build import cython
  File "C:\Users\mark\git\scikit-image\skimage\__init__.py", line 176, in <module>
    from .data import data_dir
  File "C:\Users\mark\git\scikit-image\skimage\data\__init__.py", line 15, in <module>
    from ..io import imread, use_plugin
  File "C:\Users\mark\git\scikit-image\skimage\io\__init__.py", line 11, in <module>
    from ._io import *
  File "C:\Users\mark\git\scikit-image\skimage\io\_io.py", line 4, in <module>
    from ..color import rgb2gray
  File "C:\Users\mark\git\scikit-image\skimage\color\__init__.py", line 1, in <module>
    from .colorconv import (convert_colorspace,
  File "C:\Users\mark\git\scikit-image\skimage\color\colorconv.py", line 59, in <module>
    from ..util import dtype, dtype_limits
  File "C:\Users\mark\git\scikit-image\skimage\util\__init__.py", line 12, in <module>
    from ._montage import montage
  File "C:\Users\mark\git\scikit-image\skimage\util\_montage.py", line 2, in <module>
    from .. import exposure
  File "C:\Users\mark\git\scikit-image\skimage\exposure\__init__.py", line 1, in <module>
    from .exposure import histogram, equalize_hist, \
  File "C:\Users\mark\git\scikit-image\skimage\exposure\exposure.py", line 3, in <module>
    from ..color import rgb2gray
ImportError: cannot import name 'rgb2gray'

```

File: skimage/data/__init__.py (hunk 1 above)
Comment (hmaarrfk) [reply skimage/data/__init__.py]: I think maybe the line that should be fixed is
```
    from .util.dtype import *
```
as we shouldn't be importing `*`

Comment (hmaarrfk) [reply skimage/data/__init__.py]: That said, as is, I think the tests will pass, and as such, `scikit-image`'s functionality should be restored. I'm inclined on leaving this fix to Issue #3156 as a "good first issue"

File: skimage/__init__.py [hunk 2]
```diff
@@ -56,15 +56,11 @@
 
 """
 
-import os.path as osp
 import imp
 import functools
 import warnings
 import sys
 
-pkg_dir = osp.abspath(osp.dirname(__file__))
```
Comment (soupault) [inline skimage/__init__.py]: I used `pkg_dir` couple of times to check which installation of `skimage` is currently imported, but `skimage.__file__` works equally well. :+1:

File: skimage/data/__init__.py [hunk 3]
```diff
@@ -12,13 +12,16 @@
...
 from ._binary_blobs import binary_blobs
 
-__all__ = ['load',
+import os.path as osp
+data_dir = osp.abspath(osp.dirname(__file__))
+
+__all__ = ['data_dir',
+           'load',
```
Comment (soupault) [inline skimage/data/__init__.py]: Do we really need to expose `load` here?

Comment (hmaarrfk) [reply skimage/data/__init__.py]: I don't know. It was there before.

**Comment by hmaarrfk** [comment]
@meeseeksdev backport to v0.14.x

**Pull request merged**